Sources/StreamChat/Controllers/BackgroundDatabaseObserver.swift

```diff
@@ -0,0 +1,277 @@
+//
+// Copyright © 2023 Stream.io Inc. All rights reserved.
+//
+
+import CoreData
+import Foundation
+
+class BackgroundDatabaseObserver<Item, DTO: NSManagedObject> {
```

Comment:
It looks like there are a lot of changes, but it is just the previous implementation of `BackgroundListDatabaseObserver` extracted in another place so it can be reused for Entity Observers